kafka-client/src/main/java/dev/responsive/kafka/internal/stores/PartitionedOperations.java

```diff
@@ -262,15 +270,19 @@ private static RemoteKVTable<?> createMongo(
   private static RemoteKVTable<?> createRS3(
       final ResponsiveKeyValueParams params,
       final SessionClients sessionClients,
+      final String changelogTopicName,
+      final Optional<TtlResolver<?, ?>> ttlResolver,
       final ResponsiveConfig config,
       final ResponsiveMetrics responsiveMetrics,
       final ResponsiveMetrics.MetricScopeBuilder scopeBuilder
   ) {
     return sessionClients.rs3TableFactory().kvTable(
         params.name().tableName(),
         config,
+        ttlResolver,
         responsiveMetrics,
-        scopeBuilder
+        scopeBuilder,
+        () -> numPartitionsForKafkaTopic(sessionClients.admin(), changelogTopicName)
```

Comment:
I'm using the number of kafka partitions in the changelog for the number of logical shards for now, but not sure if it we might want to derive it in some other way later on (eg the upstream kafka topic's partitions rather than changelog) ?

Reply:
for example IIRC when they scaled up the deduplicator it was much simpler to do because only the input topics (and I guess repartitions?) had to be scaled, not the changelogs. 

although...I would think Streams would check the changelog partition count and throw an error if it doesn't match the expected number based on source topic partitions...so maybe I'm just misremembering this? @agavra do you happen to recall the details here?

Reply:
IIRC the changelog topic still need to be scaled (after all, the store still needs to write into the changelog topic partition that corresponds to the source topic partition(s) the task consumes). I think we just didn't need to redistribute the data because we don't restore from 0 each time. So as long as the changelog topic is fully flushed to the remote store before the scaleup, the scaleup is safe and cheap (no redistribution).

Reply:
ah that's right I remember now. thanks for clarifying